cluster/log-dump.sh

```diff
@@ -25,67 +25,70 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
-# into a given local file ($3). Does not fail if the ssh command fails for any
-# reason, just prints an error to stderr.
-function save-log() {
-    local -r node_name="${1}"
-    local -r cmd="${2}"
-    local -r output_file="${3}"
-    if ! ssh-to-node "${node_name}" "${cmd}" > "${output_file}"; then
-        echo "${cmd} failed for ${node_name}" >&2
+    if [[ "${KUBERNETES_PROVIDER}" == "aws" ]]; then
+        local ip=$(get_instance_public_ip "${node}")
+        scp -i "${AWS_SSH_KEY}" "${SSH_USER}@${ip}:${scp_files}" "${dir}" > /dev/null || true
+    else
+        gcloud compute copy-files --project "${PROJECT}" --zone "${ZONE}" "${node}:${scp_files}" "${dir}" > /dev/null || true
     fi
 }
 
-# Saves logs common to master and nodes. The node name is in $1 and the
-# directory/name prefix is in $2. Assumes KUBERNETES_PROVIDER is set.
-function save-common-logs() {
+# Save logs for node $1 into directory $2. Pass in any non-common files in $3.
+# $3 should be a space-separated list of files.
+# This function shouldn't ever trigger errexit
+function save-logs() {
     local -r node_name="${1}"
-    local -r prefix="${2}"
-    echo "Dumping logs for ${node_name}"
-    save-log "${node_name}" "cat /var/log/kern.log" "${prefix}-kern.log"
-    save-log "${node_name}" "cat /var/log/docker.log" "${prefix}-docker.log"
+    local -r dir="${2}"
+    local files="${3} ${common_logfiles}"
     if [[ "${KUBERNETES_PROVIDER}" == "gce" ]]; then
-        save-log "${node_name}" "cat /var/log/startupscript.log" "${prefix}-startupscript.log"
+        files="${files} ${gce_logfiles}"
     fi
     if ssh-to-node "${node_name}" "sudo systemctl status kubelet.service" &> /dev/null; then
-        save-log "${node_name}" "sudo journalctl --output=cat -u kubelet.service" "${prefix}-kubelet.log"
+        ssh-to-node "${node_name}" "sudo journalctl --output=cat -u kubelet.service" > "${dir}/kubelet.log" || true
     else
-        save-log "${node_name}" "cat /var/log/kubelet.log" "${prefix}-kubelet.log"
-        save-log "${node_name}" "cat /var/log/supervisor/supervisord.log" "${prefix}-supervisord.log"
-        save-log "${node_name}" "cat /var/log/supervisor/kubelet-stdout.log" "${prefix}-supervisord-kubelet-stdout.log"
-        save-log "${node_name}" "cat /var/log/supervisor/kubelet-stderr.log" "${prefix}-supervisord-kubelet-stderr.log"
+        files="${files} ${supervisord_logfiles}"
     fi
+    copy-logs-from-node "${node_name}" "${dir}" "${files}"
 }
 
 readonly master_ssh_supported_providers="gce aws kubemark"
 readonly node_ssh_supported_providers="gce gke aws"
 
+readonly master_logfiles="kube-apiserver kube-scheduler kube-controller-manager etcd"
+readonly node_logfiles="kube-proxy"
+readonly gce_logfiles="startupscript"
+readonly common_logfiles="kern docker"
+readonly supervisord_logfiles="kubelet supervisor/supervisord supervisor/kubelet-stdout supervisor/kubelet-stderr"
+
+readonly max_scp_processes=10
```

Comment:
Why this is 10? Can't it be e.g. 25? Can you add comment explaining why you chose 10?